test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8411,108 +8436,102 @@ def is_history_enabled(self):
         jobs. Specifying an extend parameter could override
         this behavior.
-
-        :param runas: Clean the job as
-        :type runas: str or None
         """
-        delete_xt = 'force'
+        delete_xt = 'nomailforce'
         select_xt = None
         if self.is_history_enabled():
             delete_xt += 'deletehist'
             select_xt = 'x'
-        job_ids = self.select(extend=select_xt)
+        jobs = self.status(JOB, extend=select_xt)
+        job_ids = sorted(list(set([x['id'] for x in jobs])))
+        host_pid_map = {}
+        for job in jobs:
+            if 'exec_host2' in job:
+                exec_host = job['exec_host2']
+            else:
+                exec_host = job.get('exec_host', None)
+            if not exec_host or 'session_id' not in job:
+                continue
+            _host = exec_host.split('/')[0].split(':')[0]
+            if _host not in host_pid_map:
+                host_pid_map.setdefault(_host, [])
+            host_pid_map[_host].append(job['session_id'])
 
         # Turn off scheduling so jobs don't start when trying to
         # delete. Restore the orignial scheduling state
         # once jobs are deleted.
-        sched_state = {}
-        for sc in self.schedulers:
-            sched_state[sc] = self.status(
-                SCHED, {'scheduling'},
-                id=self.schedulers[sc].sc_name)[0]['scheduling']
-            # runas is required here because some tests remove current user
-            # from managers list
-            self.manager(MGR_CMD_SET, SCHED, {'scheduling': 'False'},
-                         id=self.schedulers[sc].sc_name, runas='root')
-        num_jobs = len(job_ids)
-        if num_jobs >= 100:
-            self._cleanup_large_num_jobs(job_ids, runas=runas)
-
-        if num_jobs > 0 and num_jobs < 100:
+        sched_state = []
+        scheds = self.status(SCHED)
+        for sc in scheds:
+            if sc['scheduling'] == 'True':
+                sched_state.append(sc['id'])
+                # runas is required here because some tests remove
+                # current user from managers list
+                a = {'scheduling': 'False'}
+                self.manager(MGR_CMD_SET, SCHED, a, id=sc['id'],
+                             runas='root')
+                self.expect(SCHED, a, id=sc['id'])
+        try:
+            self.deljob(id=job_ids, extend=delete_xt, runas='root', wait=False)
+        except PbsDeljobError:
+            pass
+        st = int(time.time())
+        if len(job_ids) > 100:
+            for host, pids in host_pid_map.items():
+                pids = list(set(pids))
+                chunks = [pids[i:i + 5000] for i in range(0, len(pids), 5000)]
+                for chunk in chunks:
+                    self.du.run_cmd(host, ['kill', '-9'] + chunk,
+                                    runas='root', logerr=False)
+            _msg = job_ids[-1] + ';'
+            _msg += 'Job Obit notice received has error 15001'
```

Comment:
As I said in comment in expect block we need to give some time to mom to catch with sigchild else in next test when mom restarts it will take time and that may cause problem for next test (like timeout of test or failure of start of mom because init.d only waits for 180s max for mom to shutdown gracefully)

Reply:
I understand, but before your change, we had no check for log match and we did not see any problem with timeout or failure to start mom because of init script timing out. 
Also, code on line 8502 is doing exactly similar to what you are trying to do. It is counting job_state attributes to sum up to 0. I'd suggest you retain either the log msg check or keep expect on job_state.

Reply:
@arungrover before changes to this PR it was deleting job one by one (aka one qdel per jid) + kill session_id + it was doing expect() for the queue attr to unset on per job (aka qstat per job). So cleanup_jobs was invoking 3 commands (qdel + kill + qstat) per jid and then waiting for job_stat to be 0 so it was too slow that we didn't have to wait for mom to process sigchild.
But after this PR changes we made qdel and kill command much faster + we removed qstat call as it's useless because with qdel -Wforce (as root user) server will immediately delete a job from itself so there is no point of waiting for the queue attr to unset on the job. So now we need a new check which will wait and allow some time to mom to process sigchild which is log match and then after just to ensure we do check job_state to be 0 on the server.